File: fitrbackend/src/main/java/com/example/fitrbackend/dto/CreatePlanDayExerciseRequest.java
package com.example.fitrbackend.dto;

import lombok.Getter;

@Getter
public class CreatePlanDayExerciseRequest {
    private long exerciseId;
    private long targetSets;
    private long targetReps;
    private long targetDurationSeconds;
    private float targetDistance;
    private float targetCalories;
    private Float targetWeight;

    @Override
    public String toString() {
        return "CreatePlanDayExerciseRequest{" +
                "exerciseId=" + exerciseId +
                ", targetSets=" + targetSets +
                ", targetReps=" + targetReps +
                ", targetDurationSeconds=" + targetDurationSeconds +
                ", targetDistance=" + targetDistance +
                ", targetCalories=" + targetCalories +
                ", targetWeight=" + targetWeight +
                '}';
    }
}

File: fitrfrontend/fitrfrontend/API/CreatePlanDayExerciseRequest.swift
//
//  CreatePlanDayExerciseRequest.swift
//  fitrfrontend
//
//  Created by Ambrose Blay on 2/17/26.
//

import Foundation

struct CreatePlanDayExerciseRequest: Codable {
  let exerciseId: Int64
  let targetSets: Int
  let targetReps: Int
  let targetDurationSeconds: Int
  let targetDistance: Float
  let targetCalories: Float
  let targetWeight: Float

  enum CodingKeys: String, CodingKey {
    case exerciseId = "exercise_id"
    case targetSets = "target_sets"
    case targetReps = "target_reps"
    case targetDurationSeconds = "target_duration_seconds"
    case targetDistance = "target_distance"
    case targetCalories = "target_calories"
    case targetWeight = "target_weight"
  }
}
